internal/expr/expr_test.go

```diff
@@ -258,22 +286,14 @@ func (s *ExprSuite) TestParseEmptyStringLiteral(c *C) {
 	c.Assert(err, IsNil)
 }
 
-// Detect bad escaped string literal
-func (s *ExprSuite) TestParseBadEscaped(c *C) {
-	sql := "SELECT foo FROM t WHERE x = 'O'Donnell'"
-	parser := expr.NewParser()
-	_, err := parser.Parse(sql)
-	c.Assert(err, ErrorMatches, "cannot parse expression: missing right quote in string literal")
-}
-
 func (s *ExprSuite) TestParseBadFormatInput(c *C) {
-	testListInvalidId := []string{
+	testList := []string{
 		"SELECT foo FROM t WHERE x = $Address.",
 		"SELECT foo FROM t WHERE x = $Address.&d",
 		"SELECT foo FROM t WHERE x = $Address.-",
 	}
 
-	for _, sql := range testListInvalidId {
+	for _, sql := range testList {
 		parser := expr.NewParser()
 		expr, err := parser.Parse(sql)
 		c.Assert(err, ErrorMatches, "cannot parse expression: invalid identifier near char 37")
```

Comment:
We can't be that exact because this error will be thrown when the parser gives up trying to parse the identifier. Also since its invalid, its hard to tell where it is intended to start and end, so its safest just to say near.